packages/typespec-client-generator-core/multipart_api_design.md

```diff
@@ -0,0 +1,84 @@
+This doc is to design TCGC API of multipart for language emitters.
+
+# multipart payload
+
+Here is classic payload of multipart:
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+--abcde12345
+Content-Disposition: form-data; name="id"
+Content-Type: text/plain
+
+123e4567-e89b-12d3-a456-426655440000
+--abcde12345
+Content-Disposition: form-data; name="profileImage"; filename="image1.png"
+Content-Type: image/png
+
+{…file content…}
+--abcde12345--
+```
+According to https://datatracker.ietf.org/doc/html/rfc7578, multipart request payload contains multi independent parts which could be divided into two kinds: one is non-file part which contains `required "name"/optional "Content-Type" / content`; the other one is file part which contains one more `optional "filename"`.
+
+# Current TCGC API for multipart
+Currently TCGC only has boolean flag [isMultipartFileInput](https://github.com/Azure/typespec-azure/blob/ab7a066d4ac0ae23a40f9ff8f4b6037559bda34c/packages/typespec-client-generator-core/src/interfaces.ts#L368) to distinguish file and non-file. After https://github.com/microsoft/typespec/issues/3046 complete, Typespec permit users to define more info explicitly(e.g. content-type) for each part so boolean flag is not enough.
+
+# Multipart in Typespec
+Typespec support two kinds of definition for multipart:
+
+1. Model format
+
+```
+op upload(
+  @header `content-type`: "multipart/form-data",
+  @multipartBody body: {
+    fullName: HttpPart<string>,
+    headShots: HttpPart<Image>[]
+  }
+): void;
+```
+
+2. Tuple format
+
+ ```
+op upload(
+  @header `content-type`: "multipart/form-data",
+  @multipartBody body: [
+    HttpPart<string, #{ name: "fullName" }>,
+    HttpPart<File, #{ name: "file" }>
+  ]
+): void;
+```
```

Comment:
Are 2 format equivalent?

If yes, would you use a same example (with different format)?

Reply:
make sense. 
Updated.